pkg/asset/tls/ironictls.go

```diff
@@ -0,0 +1,64 @@
+package tls
+
+import (
+	"context"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"net"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+// IronicTLSCert is the asset that generates the key/cert pair that is used for
+// enabling TLS for virtual media in ironic.
+type IronicTLSCert struct {
+	SelfSignedCertKey
+}
+
+var _ asset.WritableAsset = (*IronicTLSCert)(nil)
+
+// Dependencies returns the dependency of the the cert/key pair, which includes
+// the parent CA, and install config if it depends on the install config for
+// DNS names, etc.
+func (a *IronicTLSCert) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+	}
+}
+
+// Generate generates the cert/key pair based on its dependencies.
+func (a *IronicTLSCert) Generate(ctx context.Context, dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+	hostname := internalAPIAddress(installConfig.Config)
+
+	cfg := &CertCfg{
+		Subject:      pkix.Name{CommonName: "ironic", OrganizationalUnit: []string{"openshift"}},
+		ExtKeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
+		Validity:     ValidityOneDay,
+	}
+
+	if installConfig.Config.Platform.BareMetal != nil {
+		vips := installConfig.Config.BareMetal.APIVIPs
+		if len(installConfig.Config.BareMetal.BootstrapProvisioningIP) > 0 {
```

Comment:
nit: if it's a string, it's more natural to compare it to "" rather than using len()

Reply:
Fixed